packages/clay-core/src/table/Cell.tsx

```diff
@@ -212,22 +214,29 @@ export const Cell = React.forwardRef<HTMLTableCellElement, Props>(
 					<span className="text-truncate-inline">
 						<span className="text-truncate">{children}</span>
 					</span>
-				) : treegrid && index === 0 ? (
+				) : treegrid && index === 0 && !isHead ? (
 					<Layout.ContentRow
 						style={{
 							paddingLeft:
-								(level - (expandable ? 1 : 0)) * 28 -
-								(expandable ? 4 : 0),
+								(level - (expandable || onLoadMore ? 1 : 0)) *
+									28 -
+								(expandable || onLoadMore ? 4 : 0),
 						}}
 					>
-						{expandable && (
+						{(expandable || onLoadMore) && !isLoading && (
```

Comment:
This seems to indicate that if this was not expandable but onLoadMore was passed as a prop that we would render the button used for expanding the row. What is the use case for this? I would think we would still only want to render the expand button if it was expandable.

Reply:
Yeah that's a big dilemma in this implementation and it's complicated, for example to say that an item is expandable we need the key for example `children` to be visible, so if it is not visible we make the assumption that it is not expandable in the default behavior when you have asynchronous loading the assumption we could make is if there is `children` but the array is empty we would consider it lazy loading but that is an assumption that we cannot guarantee is true for all use cases because the data structure in the DXP for example is different so it is difficult to make this assumption, so what we do here is always render the expandable when `onLoadMore` is defined but this is also not very interactive.

An idea I thought about this is we can add a property called ` lazy` in the row to say that we should consider asynchronous loading so the developer can use any conditional he wants according to his data structure would be more flexible and we avoid assumptions.
```jsx
<Table
	nestedKey="items"
	onLoadMore={async (item: any) => {
		// load more
	}}
>
	<Head items={columns}>
		{(column) => (
			<Cell className="table-cell-minw-300" key={column.id}>
				{column.name}
			</Cell>
		)}
	</Head>

	<Body defaultItems={items}>
		{(row) => (
			<Row lazy={row.items.lenght === 0}>
				...
			</Row>
		)}
	</Body>
</Table>
```

Let me know what you think.